mypy/stubtest.py

```diff
@@ -1010,33 +1011,71 @@ def verify_typealias(
     stub: nodes.TypeAlias, runtime: MaybeMissing[Any], object_path: List[str]
 ) -> Iterator[Error]:
     stub_target = mypy.types.get_proper_type(stub.target)
+    stub_desc = f"Type alias for {stub_target}"
     if isinstance(runtime, Missing):
-        yield Error(
-            object_path, "is not present at runtime", stub, runtime,
-            stub_desc=f"Type alias for: {stub_target}"
-        )
+        yield Error(object_path, "is not present at runtime", stub, runtime, stub_desc=stub_desc)
         return
+    runtime_origin = get_origin(runtime) or runtime
     if isinstance(stub_target, mypy.types.Instance):
-        yield from verify(stub_target.type, runtime, object_path)
+        if not isinstance(runtime_origin, type):
+            yield Error(
+                object_path,
+                "is inconsistent, runtime is not a type",
+                stub,
+                runtime,
+                stub_desc=stub_desc,
+            )
+            return
+
+        stub_origin = stub_target.type
+        # Do our best to figure out the fullname of the runtime object...
+        runtime_name: object
+        try:
+            runtime_name = runtime_origin.__qualname__
+        except AttributeError:
+            runtime_name = getattr(runtime_origin, "__name__", MISSING)
+        if isinstance(runtime_name, str):
+            runtime_module: object = getattr(runtime_origin, "__module__", MISSING)
+            if isinstance(runtime_module, str):
+                if runtime_module == "collections.abc" or (
+                    runtime_module == "re" and runtime_name in {"Match", "Pattern"}
+                ):
+                    runtime_module = "typing"
+                runtime_fullname = f"{runtime_module}.{runtime_name}"
+                if re.match(fr"_?{re.escape(stub_origin.fullname)}", runtime_fullname):
```

Comment:
Should be fullmatch? Or is that intentional?

Reply:
Not intentional — fullmatch is better here, yup!

```suggestion
                if re.fullmatch(fr"_?{re.escape(stub_origin.fullname)}", runtime_fullname):
```